example-app/app/core/components/sidenav.component.ts

```diff
@@ -19,4 +24,10 @@ import { Component, Input } from '@angular/core';
 })
 export class SidenavComponent {
   @Input() open = false;
+
+  constructor(private store: Store<fromRoot.State>) {}
+
+  closeSidenav() {
+    this.store.dispatch(new LayoutActions.CloseSidenav());
```

Comment:
Following the dump/smart component convention we can't `dispatch` inside this pure component,
we have to use an `@Output` property to emit the close event to the container page located at `example-app/app/core/containers/app.component`.

Reply:
Thanks @timdeschryver, I will change this according to your suggestions.